app/api-v1/services/orderService.js

```diff
@@ -1,13 +1,15 @@
 const { postOrderDb, getRecipeByIDs } = require('../../db')
-const { BadRequestError, IncorrectSupplierError } = require('../../utils/errors')
+const { BadRequestError, IncorrectSupplierError, SupplierDoesNotExistError } = require('../../utils/errors')
 
 async function postOrder(reqBody) {
   // Will add a get function at a later date to check for duplication
 
   // This section checks if the order manufacturer does not match the supplier
   const uniqueRecipeIDs = [...new Set(reqBody.items)]
   const recipes = await getRecipeByIDs(uniqueRecipeIDs)
-  if (recipes) {
+  if (recipes.length == 0) {
+    throw new SupplierDoesNotExistError({ message: 'Order post error - Supplier does not exist', service: 'order' })
+  } else {
```

Comment:
Hmm not sure this is the right check. I think we want to check if `uniqueRecipeIDs.length !== recipes.length`. Then if not its a `RecipeDoesNoExistError` rather than supplier.